flake/packages/neovim.nix

```diff
@@ -0,0 +1,90 @@
+{
+  neovim-src,
+  lib,
+  pkgs,
+  ...
+}: let
+  src = pkgs.fetchFromGitHub {
+    owner = "neovim";
+    repo = "neovim";
+    inherit (neovim-src) rev;
+    hash = neovim-src.narHash;
+  };
+
+  deps = lib.pipe "${src}/cmake.deps/deps.txt" [
+    builtins.readFile
+    (lib.splitString "\n")
+    (map (builtins.match "([A-Z0-0_]+)_(URL|SHA256)[[:space:]]+([^[:space:]]+)[[:space:]]*"))
+    (lib.remove null)
+    (lib.flip builtins.foldl' {}
+      (acc: matches: let
+        name = lib.toLower (builtins.elemAt matches 0);
+        key = lib.toLower (builtins.elemAt matches 1);
+        value = lib.toLower (builtins.elemAt matches 2);
+      in
+        acc
+        // {
+          ${name} =
+            acc.${name}
+            or {}
+            // {
+              ${key} = value;
+            };
+        }))
+    (builtins.mapAttrs (lib.const pkgs.fetchurl))
+  ];
+in
+  (pkgs.neovim-unwrapped.override {
+    gettext = pkgs.gettext.overrideAttrs (_: {
+      src = deps.gettext;
+    });
+    libiconv = pkgs.libiconv.overrideAttrs (_: {
+      src = deps.libiconv;
+    });
+    libuv = pkgs.libuv.overrideAttrs (_: {
+      src = deps.libuv;
+    });
+    libvterm-neovim = pkgs.libvterm-neovim.overrideAttrs (_: {
+      src = deps.libvterm;
+    });
+    msgpack-c = pkgs.msgpack-c.overrideAttrs (_: {
+      src = deps.msgpack;
+    });
+    tree-sitter = pkgs.tree-sitter.override (_: {
+      rustPlatform =
+        pkgs.rustPlatform
+        // {
+          buildRustPackage = args:
+            pkgs.rustPlatform.buildRustPackage (args
+              // {
+                src = deps.treesitter;
+                cargoHash = "sha256-U2YXpNwtaSSEftswI0p0+npDJqOq5GqxEUlOPRlJGmQ=";
+              });
+        };
+    });
+    treesitter-parsers = let
+      grammars = lib.filterAttrs (name: _: lib.hasPrefix "treesitter_" name) deps;
+      parsers =
+        lib.mapAttrs'
+        (name: value: lib.nameValuePair (lib.removePrefix "treesitter_" name) {src = value;})
```

Comment:
s/{src = value;}/value

to fix the build error you're seeing

Reply:
Hmm, kinda screwed with bundling the markdown parsers because the reverted commit removes the ability to specify a sourceRoot. So maybe pin nixpkgs at https://github.com/NixOS/nixpkgs/commit/db2ee055124d9714712d703d401d586cf6e82e67?

Looks like these reverts are temporary until the breaking change window opens in nixpkgs so I think the move is to just pin nixpkgs at that commit (it is the parent of the reverted commit) and then once the 0.10 is merged back into nixpkgs we can unpin it

Reply:
and just to confirm: build succeeds after pinning nixpkgs